src/main/java/nl/tudelft/contextproject/controller/GameController.java

```diff
@@ -109,7 +117,122 @@ public void onAction(String name, boolean isPressed, float tpf) {
+		keypicblue.setWidth(width / 30);
+		keypicblue.setHeight(height / 12);
+		keypicblue.setPosition(width * 0.6f, 0);
+		addGuiElement(keypicyellow);
+		addGuiElement(keypicred);
+		addGuiElement(keypicblue);
+		//attach the bombcounter
+		BitmapText hudTextb = new BitmapText(Main.getInstance().getGuiFont(), false);
+		hudTextb.setSize(height / 30);
+		hudTextb.setColor(ColorRGBA.White);
+		hudTextb.setText("" + Main.getInstance().getCurrentGame().getPlayer().getInventory().numberOfBombs());
+		hudTextb.setLocalTranslation(width / 3f, hudTextb.getLineHeight() + height / 40, 0);
+		addGuiElement(hudTextb);
+		//attach bombicon
+		Picture pic = new Picture("Bomb Picture");
+		pic.setImage(Main.getInstance().getAssetManager(), "Textures/bombicon.png", true);
+		pic.setWidth(width / 15);
+		pic.setHeight(height / 10);
+		pic.setPosition(width / 4f, 0);
+		addGuiElement(pic);
+		
+		//attach your healthcounter        
+		Picture heart1 = healthContainer(1, height, width);
+		addGuiElement(heart1);
+		Picture heart2 = healthContainer(2, height, width);
+		addGuiElement(heart2);
+		Picture heart3 = healthContainer(3, height, width);
+		addGuiElement(heart3);
+		//update the gui
+		Main.getInstance().attachTickListener(new TickListener() {
+			
+			@Override
+			public void update(float tpf) {
+				//bomb update
+				hudTextb.setText("" + Main.getInstance().getCurrentGame().getPlayer().getInventory().numberOfBombs()); 
+				
+				//health update
+				if (game.getPlayer().getHealth() <= 2.5) {
+					heart3.setImage(Main.getInstance().getAssetManager(), "Textures/emptyheart.png", true);
+				}
+				if (game.getPlayer().getHealth() <= 1.5) {
+					heart2.setImage(Main.getInstance().getAssetManager(), "Textures/emptyheart.png", true);
+				}
+				if (game.getPlayer().getHealth() <= 0.5) {
+					heart1.setImage(Main.getInstance().getAssetManager(), "Textures/emptyheart.png", true);
+				}
+				//keys update
+				if ((game.getPlayer().getInventory().containsColorKey(ColorRGBA.Yellow.set(1.0f, 1.0f, 0.0f, 0.0f)))) {
+					keypicyellow.setImage(Main.getInstance().getAssetManager(), "Textures/yellowkeyicon.png", true);
+				} else {
+					keypicyellow.setImage(Main.getInstance().getAssetManager(), "Textures/emptykeyicon.png", true);
+				}
+				if ((game.getPlayer().getInventory().containsColorKey(ColorRGBA.Blue.set(0.0f, 0.0f, 1.0f, 0.0f)))) {
+					keypicblue.setImage(Main.getInstance().getAssetManager(), "Textures/bluekeyicon.png", true);
+				} else {
+					keypicblue.setImage(Main.getInstance().getAssetManager(), "Textures/emptykeyicon.png", true);
+				}
+				if ((game.getPlayer().getInventory().containsColorKey(ColorRGBA.Red.set(1.0f, 0.0f, 0.0f, 0.0f)))) {
+					keypicred.setImage(Main.getInstance().getAssetManager(), "Textures/redkeyicon.png", true);
+				} else {
+					keypicred.setImage(Main.getInstance().getAssetManager(), "Textures/emptykeyicon.png", true);
+				}
+			}
```

Comment:
We need to split up this method in smaller pieces, but we should do that in the next sprint.